src/sentry/tagstore/v2/backend.py

```diff
@@ -356,43 +427,63 @@ def incr_group_tag_key_values_seen(self, project_id, group_id, environment_id, k
                         'project_id': project_id,
                         'group_id': group_id,
                         'environment_id': environment_id,
-                        'key': key,
+                        '_key_id': key,
                     })
 
     def incr_group_tag_value_times_seen(self, project_id, group_id, environment_id,
                                         key, value, extra=None, count=1):
-        buffer.incr(GroupTagValue,
-                    columns={
-                        'times_seen': count,
-                    },
-                    filters={
-                        'project_id': project_id,
-                        'group_id': group_id,
-                        'environment_id': environment_id,
-                        'key': key,
-                        'value': value,
-                    },
-                    extra=extra)
+        for env in [environment_id, None]:
+            tagkey, _ = self.get_or_create_tag_key(project_id, env, key)
+            tagvalue, _ = self.get_or_create_tag_value(project_id, env, key, value)
+
+            buffer.incr(GroupTagValue,
+                        columns={
+                            'times_seen': count,
+                        },
+                        filters={
+                            'project_id': project_id,
+                            'group_id': group_id,
+                            'environment_id': env,
+                            '_key_id': tagkey.id,
+                            '_value_id': tagvalue.id,
+                        },
+                        extra=extra)
 
     def get_group_event_ids(self, project_id, group_id, environment_id, tags):
```

Comment:
I'm going to come back to this method a little later and think through it harder.